drivers/soundwire/intel_init.c

```diff
@@ -239,41 +235,46 @@ static struct sdw_intel_ctx
 
 	INIT_LIST_HEAD(&ctx->link_list);
 
-	ret = driver_register(sdw_intel_link_ops.driver);
-	if (ret) {
-		dev_err(&adev->dev, "failed to register sdw master driver: %d\n", ret);
-		goto register_err;
-	}
-
 	/* Create SDW Master devices */
 	for (i = 0; i < count; i++, link++) {
-		if (link_mask && !(link_mask & BIT(i)))
+		if (!(link_mask & BIT(i))) {
```

Comment:
Should we return error or do something if link_mask = 0?

Reply:
the mask is generated in SOF and we print debug message there. if we want to check it , It is better to do it at root node, or it will be checked many times.

Reply:
> Should we return error or do something if link_mask = 0?

yes we could good suggestion.

Reply:
> the mask is generated in SOF and we print debug message there. if we want to check it , It is better to do it at root node(SOF), or it will be checked many times.

I don't understand the comment @RanderWang . If you want to provide an error message for each link, then this is the right place to do so. If you have another idea please suggest a patch